### PR TITLE
feat(batcher client): add verify function in service manager to check proof merkle path

### LINKED ISSUE #320: feat: add verify function in service manager to check proof merkle path

### PR DESCRIPTION
**Pull request by entropidelic**
Closes #320 

A new function has been added to the aligned service manager contract, `verifyBatchInclusion`. This function receives what is responded to the Aligned users when they send their proofs to the batcher and verifies that the proof sent to Aligned is included in a verified batch.

To this end, the batcher client has been refactored to admit two subcommands: `submit`, to send the proofs just as we have been doing up to now, and `verify-proof-onchain`, to use the new contract feature.

To test it in local devnet:
* Start anvil with block time, operator, aggregator and batcher.
* To test it just as a user will use it, run `make install_aligned_compiling`.
* submit proofs with
```
aligned submit --proving_system GnarkPlonkBls12_381 \
--proof batcher/aligned/test_files/plonk_bls12_381/plonk.proof \
--public_input batcher/aligned/test_files/plonk_bls12_381/plonk_pub_input.pub \
--vk batcher/aligned/test_files/plonk_bls12_381/plonk.vk \
--repetitions 4
```
when responses are received from the batcher, each one will be written into a json file inside the `aligned_verification_data` folder with the format `{batch_merkle_root}_{proof_index_in_batch}.json`, for example `19f04bbb143af72105e2287935c320cc2aa9eeda0fe1f3ffabbe4e59cdbab691_0.json`.
* To verify that a proof was included in the batch, run
```
aligned verify-proof-onchain \
--aligned-verification-data aligned_verification_data/19f04bbb143af72105e2287935c320cc2aa9eeda0fe1f3ffabbe4e59cdbab691_0.json \
--private-key-store config-files/anvil.ecdsa.key.json
```
Note that your JSON batch inclusion response may or may not have the same name as this example. If everything works correctly, you should get the response 
```
[2024-06-12T18:09:30Z INFO  aligned] Batch inclusion verification response: true
```
You can try the same with the other responses you received too.

**Bonus points**: You can also try changing some data of these JSON batch inclusion data and test the `verify-inclusion` command. In this case, it should output `false`.

### Holesky testing
To test in holesky, be sure to add the `--chain holesky` argument to the `verify-inclusion` subcommand. For example, 
```
aligned verify-proof-onchain \
--aligned-verification-data aligned_verification_data/681fcf9f64e7451a7f250e7945cef644ed52f6efccaec1a3294595405042009c_0.json \
--private-key-store config-files/anvil.ecdsa.key.json \
--chain holesky
```

### CODE REVIEW

**Review by jotabulacios — DISMISSED** [review]
lgtm.
verified all proofs, and they failed as expected when data changed

**Review by NicolasRampoldi — DISMISSED** [review]
lgtm, tested and it works. I made an issue #377 related to the expect in the verify function but is not related to this PR, it was from before.

**Review by NicolasRampoldi — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -111,14 +113,32 @@ aligned \
...
 ```
 
+### Verify proof batch inclusion on Ethereum
+After running the commands of the previous section to submit proofs to the batcher, you will receive responses that will be written to disk in a JSON format inside the `<batch_inclusion_data_directory_path>`, for example `19f04bbb143af72105e2287935c320cc2aa9eeda0fe1f3ffabbe4e59cdbab691_0.json`. By default, the `batch_inclusion_data` directory will be created where the submit command is being executed, but you can specify it with the `<batch_inclusion_data_directory_path>` argument. To verify their inclusion in a batch, run the following command, replacing the `<path_to_batch_inclusion_data>` placeholder with the path to your response file, and `<private_key_store>` with the path to your ECDSA key store:
+
+```bash
+aligned verify-inclusion \
+--inclusion-data <path_to_batch_inclusion_data> \
+--private-key-store <path_to_private_key_store> \
+--chain holesky
```
Comment (MauroToscano) [inline README.md]:
```suggestion
aligned verify-proof-onchain \
--aligned-verification-data <path_to_batch_verification_data> \
--private-key-store <path_to_private_key_store> \
--chain holesky
```

**Review by taturosati — APPROVED** [review]
lgtm

File: README.md [hunk 2]
```diff
@@ -111,14 +113,35 @@ aligned \
...
-aligned \
+aligned submit \
 --proving_system Groth16Bn254 \
 --proof ./batcher/aligned/test_files/groth16/ineq_1_groth16.proof \
 --public_input ./batcher/aligned/test_files/groth16/ineq_1_groth16.pub \
 --vk ./batcher/aligned/test_files/groth16/ineq_1_groth16.vk \
 --conn wss://batcher.alignedlayer.com
 ```
 
+### Verify proof batch inclusion on Ethereum
```
Comment (MauroToscano) [inline README.md]:
```suggestion
### Creating a transaction from the CLI to verify proof in Ethereum
```

File: contracts/src/core/AlignedLayerServiceManager.sol [hunk 3]
```diff
@@ -116,4 +114,41 @@ contract AlignedLayerServiceManager is ServiceManagerBase, BLSSignatureChecker,
...
+        bytes32 pubInputCommitment,
+        bytes32 provingSystemAuxDataCommitment,
+        bytes20 proofGeneratorAddr,
+        bytes32 batchMerkleRoot,
+        bytes memory merkleProof,
+        uint verificationDataBatchIndex
+    ) external returns (bool) {
+        require(
+            batchesState[batchMerkleRoot].taskCreatedBlock != 0,
+            "Batch doesn't exists"
```
Comment (uri-99) [inline contracts/src/core/AlignedLayerServiceManager.sol]:
```suggestion
            "Batch doesn't exist"
```

**Review by uri-99 — CHANGES_REQUESTED** [review]

**Review by uri-99 — APPROVED** [review]

**Review by JuArce — APPROVED** [review]

**Pull request merged**